### DISCUSSION TITLE
Planning 9.0.1.6 release

### DISCUSSION THREAD

**Discussion by BobanSpasic** [bgrabitmap/bgracontrols#190] [General]
Hi guys.
Before the release, I need to fix some bugs in TBCLea components.
I have a bug in range calculation in BCRingSlider component, and a background color bug in BCLeaBoard.
Please, wait for me before release.

**Comment by lainz** [comment]
Ok no problem. When ready you can make the release. Don't forget to edit
the OPM package file version.

**Comment by BobanSpasic** [comment]
Done with the bugfixes. Committed to the dev branch.
As for the release - I am afraid of doing it. I'm not very versed with git. I know just the basics.

**Comment by lainz** [comment]
Just try you will not break nothing. You don't need to know git, just use GitHub website.

**Comment by BobanSpasic** [comment]
Well... it is gone wrong.
I am not doing it anymore... ever.
OPM have downloaded older version instead of 9.0.1.6

**Comment by lainz** [comment]
Hi. Yes that's normal. Because we need to inform to GetMem the new version
and he adds it to OPM

**Comment by lainz** [comment]
Hi Boban. Please don't give up. There's a good opportunity for you to do
the release.

Maybe I didn't explain it well. The OPM json is for other thing. But to
properly release for OPM we need to send an email to GetMem.

So I think you did well the release. Sorry for not explaining before all
the steps. If you want send me a Pm and I will send you the GetMem mail. So
all the steps are done.

I don't know if you reverted the release. Just do it again and theres no
problem about the git history it just doesn't matter.

Hope you don't feel discouraged to contribute to open source.

Best regards
Lainz

**Comment by BobanSpasic** [comment]
Well, I've updated the version in the json file and did a new release with a new tag.
After that I've checked it in OPM and I saw that the versions in lpk files also need to be updated.
After the update, I saw that an older package has been installed.
At this point, I've decided to delete the release and revert the version number in the json file.
My OPM still says that 9.0.1.6 is available, even after the json file is saying 9.0.1.5.
So, at this point, it was clear for me that I've messed up:
- OPM is showing a non-existing release
- do the LPK files needs an updated version info or not - it is something that I've got to ask before pushing a release, but I wasn't aware of it

**Comment by lainz** [comment]
Hi, yes the lpk needs and updated info, as well the json.

The thing is that the updates.json for opm was working before, and it was downloading from author website, but for me doesn't work as well. Is not an issue with your release.

The thing is do the release, and send a mail to "getmem1[at]gmail[dot]com" so he updates BGRAControls in OPM repository.

**Comment by BobanSpasic** [comment]
The link in json is referring to a non-existing URL.
Maybe the link should also be updated.

**Comment by lainz** [comment]
Done. Changed to
https://github.com/bgrabitmap/bgracontrols/archive/refs/heads/master.zip

**Comment by BobanSpasic** [comment]
I just merged the dev branch to master, but the json file didn't got updated with new version strings 9.0.1.6

**Comment by lainz** [comment]
Thanks, all looks good for me.

Can you contact getmem? Send him an email saying you did the new bgracontrols release and want to add to OPM. Send a link of the release, this link:

https://github.com/bgrabitmap/bgracontrols/releases/tag/v9.0.1.6

And post a message in the BGRAControls thread,

Many thanks for your help!!!

=)

**Comment by BobanSpasic** [comment]
Done and done

**Comment by lainz** [comment]
Thanks!! =)

**Comment by BobanSpasic** [comment]
Well, this didn't go smooth...
GetMem isn't responding to the emails (not only from me, but in general). I hope he is doing well.  
wp is in charge of the OPM in the GetMem's absence.  
I've contacted wp about updating the OPM, and there are some problems on our side.  
As I've understood, the GetMem was the one who packaged the BGRAControls for OPM in the past, right? Now, I wouldn't ask wp to do the same as GetMem did, and that's why we are the ones who need to properly prepare the package for the release in OPM. This means a bit of cleaning in the repository.  
Thoughts?

**Reply by wp-xyz** [reply]
@lainz: 

Please understand that there are more than 200 OPM packages, and I am not planning to understand and test each of these packages in every detail. The more well-prepared a package is the smaller is the chance that I do something wrong.

When preparing the new BGRAControls for OPM I noticed that
- the meta data (Lazarus compatibility, FPC compatibility, Supported widgetsets) seem to be out of date, and asked for an update, at least as far as you know (please don't expect me to test the package on every widgetset and with every Laz/FPC combination).
- the zip contains internal git files (.git, .gitignore, -config.yml) which are of no use for the user. Please remove them. Of course I can do that myself, but it is likely that I'll forget it occasionally
- the zip contains the package lazmousekeyinput which is distributed along with Lazarus. GetMem's package generator for OPM detects this lpk file and wants to install it along with the BGRAControls packages. I guess that this is not intended, and probably it is even harmful. Again, please don't want me to investigate why this package is there, you know better than me. The Lazarus way is to add a needed package to the requirements of a package or (sample) project. Or, if you only want to simplify tests by adding these units to the search path, remove at least the lpk file so that OPM does not try to install it.

**Reply by lainz** [reply]
Hi to all. At the moment Im not interested in making bgracontrols. Im not using it as currently Im doing Android apps.

Some of the other developers listed in bgracontrols please take care of the package.

Thanks.

Hope someone can keep it alive.

**Reply by BobanSpasic** [reply]
Anyone with a good understanding of the BGRAControls code and the repository structure here?
Any volunteers?

**Reply by lainz** [reply]
I think you Boban are the indicated one to keep this project alive. You already know how to do the release.

If not available in OPM is not a big deal because one can install any package from the repositories at github just downloading them and just installing the lpk.

My main concern is not OPM but the keep on making releases with fixes and new controls.

Thats my point of view.
